code/src/sixsq/nuvla/server/resources/spec/vulnerability.cljc

```diff
@@ -0,0 +1,59 @@
+(ns sixsq.nuvla.server.resources.spec.vulnerability
+  (:require
+    [clojure.spec.alpha :as s]
+    [sixsq.nuvla.server.resources.spec.common :as common]
+    [sixsq.nuvla.server.util.spec :as su]
+    [sixsq.nuvla.server.resources.spec.core :as core]
+    [spec-tools.core :as st]))
+
+(s/def ::reference
+  (-> (st/spec ::core/nonblank-string)
+    (assoc :name "reference"
+           :json-schema/type "string"
+
+           :json-schema/description "reference to the official vulnerability page"
+           :json-schema/group "body"
+           :json-schema/order 18)))
+
+(s/def ::published
+  (-> (st/spec ::core/nonblank-string)
+    (assoc :name "published"
+           :json-schema/display-name "publish date"
+           :json-schema/description "date when the vulnerability was officially published"
+
+           :json-schema/group "body"
+           :json-schema/order 19)))
+
+(s/def ::modified
+  (-> (st/spec ::core/nonblank-string)
+    (assoc :name "modified"
+           :json-schema/display-name "modified date"
+           :json-schema/description "date when the vulnerability was officially modified"
+
+           :json-schema/group "body"
+           :json-schema/order 20)))
+
+(s/def ::score
+  (-> (st/spec (s/and number? #(not (neg? %))))
+    (assoc :name "score"
+
```

Comment:
@0xbase12 the testing issue I've mentioned a few days ago can be achieved by changing this line, to indicate an attr type, like "string" (which is wrong). The tests will then complain about something completely unrelated